viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -0,0 +1,284 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { Cognite3DViewer, Intersection } from '@reveal/core';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+import * as THREE from 'three';
+import { MeasurementLabel } from './MeasurementLabel';
+import { MeasurementLineOptions, MeasurementOptions, MeasurementLabelUpdateDelegate } from './types';
+import { LineGeometry } from 'three/examples/jsm/lines/LineGeometry';
+import { LineMaterial } from 'three/examples/jsm/lines/LineMaterial';
+
+export class MeasurementTool extends Cognite3DViewerToolBase {
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _measurementLabel: MeasurementLabel;
+  private _lineGeometry: LineGeometry;
+  private _lineMesh: THREE.Mesh;
+  private _lineMaterial: LineMaterial;
+  private readonly _linePosition: Float32Array;
+  private readonly _options: MeasurementOptions;
+  private _pointSize: number;
+  private _distanceValue: string;
+  private _distanceToCamera: number;
+
+  private readonly _handleonPointerClick = this.onPointerClick.bind(this);
+  private readonly _handleonPointerMove = this.onPointerMove.bind(this);
+
+  private readonly _lineOptions: MeasurementLineOptions = {
+    lineWidth: 0.002,
+    color: 0x00ffff
+  };
+
+  constructor(viewer: Cognite3DViewer, options?: MeasurementOptions) {
+    super();
+    this._viewer = viewer;
+    this._options = options ?? {};
+    this._measurementLabel = new MeasurementLabel(this._viewer);
+    this._linePosition = new Float32Array(6);
+    this._pointSize = 0.02;
+    this._distanceValue = '';
+    this._distanceToCamera = 0;
+  }
+
+  /**
+   * Add an measurement.
+   */
+  add(): void {
+    this.setupEventHandling();
+  }
+
+  /**
+   * Remove measurement.
+   */
+  remove(): void {
+    //remove measurement label, clear all mesh, geometry & event handling.
+    this._measurementLabel.remove();
+    this.clearLineObjects();
+    this.removeEventHandling();
+  }
+
+  /**
+   * Update the measurement line width.
+   * @param width Line width
+   */
+  updateLineWidth(width: number): void {
+    this._lineOptions.lineWidth = width;
+    this._pointSize = width * 10.0;
+  }
+
+  /**
+   * Update measurement line color.
+   * @param color Line color.
+   */
+  updateLineColor(color: number): void {
+    this._lineOptions.color = color;
+  }
```

Comment:
Replace with `setLineOptions(options: MeasurementLineOptions)` that overrides current settings with the provided settings ("merges" current with fields from the updated optioms)